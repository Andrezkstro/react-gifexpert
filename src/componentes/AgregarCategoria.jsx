import { useState} from 'react'
export const AgregarCategoria = ({ onNewCategoria }) => {

    const [inputValor, setinputValor] = useState("")

    const cambiarImput = ({target}) => {
        setinputValor(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValor.trim().length <= 1) return;

        onNewCategoria( inputValor.trim() );
        //cambiarCategorias(categorias => [inputValor, ...categorias])
        setinputValor('');
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValor}
                onChange={ cambiarImput }
            />
        </form>
    )
}
