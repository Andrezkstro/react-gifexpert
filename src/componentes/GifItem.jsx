import PropTypes from 'prop-types';

export const GifItem = ({titulo, url, id}) => {
    return (
        <div className="card">
            <img src={ url } alt={ titulo } />
            <p> { titulo } </p>
        </div>
    )
}


GifItem.propTypes = {
    titulo: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}