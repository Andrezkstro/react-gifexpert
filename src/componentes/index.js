export * from './AgregarCategoria'
export * from './GifGrid'
export * from './GifItem'