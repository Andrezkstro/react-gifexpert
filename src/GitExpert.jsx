import { useState } from 'react'
import { AgregarCategoria, GifGrid } from './componentes';

export const GitExpert = () => {
    const [categorias, setCategorias] = useState(['One Piece']);

    const onAddCategoria = (newCategoria) => {
        if (categorias.includes(newCategoria)) return;

        setCategorias([newCategoria, ...categorias])
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AgregarCategoria
                onNewCategoria={onAddCategoria}
            />

            {categorias.map((categoria) => (
                <GifGrid key={categoria} categoria={categoria} />
            ))
            }
        </>
    )
}