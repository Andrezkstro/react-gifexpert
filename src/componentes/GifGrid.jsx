import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ categoria }) => {

    const {imagenes, isLoading } = useFetchGifs (categoria);

    return (
        <>
            <h3>{categoria}</h3>
            {
                isLoading && (<h2> Cargando ......</h2>)
            }
            <div className="card-grid">
                {imagenes.map((image) => (
                    <GifItem
                        key={ image.id }
                        { ...image }
                    />
                ))
                }
            </div>
        </>
    )

}
