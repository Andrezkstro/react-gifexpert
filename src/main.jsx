import React from 'react';
import ReactDOM from 'react-dom/client';
import { GitExpert } from './GitExpert';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GitExpert />
  </React.StrictMode>
)
