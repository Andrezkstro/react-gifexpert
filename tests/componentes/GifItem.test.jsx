import { render } from "@testing-library/react";
import { GifItem } from "../../src/componentes";

describe('Pruebas GifItem', () =>{
    const titulo = 'Saitama';
    const url    = 'https://saitama.com'
    test('hacer match ', () => {
        const { container } = render( <GifItem titulo= {titulo} url={url}/>)
        expect( container ).toMatchSnapshot();
    });
});