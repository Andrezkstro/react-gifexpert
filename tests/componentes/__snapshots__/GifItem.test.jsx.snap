// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifItem hacer match  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com"
    />
    <p>
       
      Saitama
       
    </p>
  </div>
</div>
`;
